src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -518,6 +518,10 @@ bool C2Compiler::is_intrinsic_supported(const methodHandle& method, bool is_virt
   case vmIntrinsics::_fsignum:
     if (!Matcher::match_rule_supported(Op_SignumF)) return false;
     break;
+  case vmIntrinsics::_VectorComExp:
```

Comment:
Why `_VectorComExp` intrinsic is special? Other vector intrinsics are handled later and in a different manner.

What about `ExpandV` case?

Reply:
It was an attempt to facilitate in-lining of these APIs over targets which do not intrinsify them. I agree its not a generic fix since three APIs are piggybacking on same entry point and without the knowledge of opcode it will be inappropriate to take any call at this place, lazy intrinsification gives opportunity for some of the predications to concertize as compilation happens under closed world assumptions.

Reply:
Still not clear why the code is shaped the way it is.

`Matcher::match_rule_supported_vector()` already checks that there are relevant matching rules.

The checks require both `CompressM` and `CompressV` to be present to enable the intrinsic. Is it important?

Also, it doesn't take `EnableVectorSupport` into account while all other vector intrinsics respect it.